src/App.tsx

```diff
@@ -1,109 +1,40 @@
 import React from 'react';
 
 import './App.scss';
-
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import { PostList } from './components/PostList';
+
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+import { User } from './types/User';
+import { Comment } from './types/Comment';
+import { Post, PostFromServer } from './types/Post';
+
+export function getPostByID(
+  posts: Post[],
+  users: User[],
+  commentsData: Comment[],
+): PostFromServer[] {
+  return posts.map((post) => ({
+    ...post,
+    user: users.find((user) => (
+      user.id === post.userId
+    )) || null,
+    comments: commentsData.filter((comment) => (
+      comment.postId === post.id
+    )) || null,
+  }));
+}
```

Comment:
- You have to separate these operations into separate functions

Reply:
I made some changes, but did not separate the functions. I solved this task as Yura showed in class, he said that this code is ok. I've tried splitting functions, but I get too much code or errors.